warehouse/packaging/models.py

```diff
@@ -265,6 +268,15 @@ def __table_args__(cls):  # noqa
     )
     classifiers = association_proxy("_classifiers", "classifier")
 
+    @property
+    def structured_classifiers(self):
+        structured = defaultdict(list)
+        for classifier in self.classifiers:
+            key, *value = classifier.split(' :: ', 1)
+            if value:
+                structured[key].append(value[0])
+        return structured
```

Comment:
It's almost the same, except I need the tail, not the whole classifier, as the value. I don't see a _great_ way of unifying them right now.